### PR TITLE
port: [#4163] log multi-choice options in App Insights

### LINKED ISSUE #4163: port: Bug fix: multi-choice options can now be logged in App Insights (#6261)
The changes in [Bug fix: multi-choice options can now be logged in App Insights (#6261)](https://github.com/microsoft/botbuilder-dotnet/pull/6261) may need to be ported to maintain parity with `microsoft/botbuilder-dotnet`.

<blockquote>
Fixes #6262 

## Description

Fixed multi-choice prompt options bug not being logged by Application Insights. 

## Specific Changes

  - Created a virtual method in [/InputDialog.cs](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/InputDialog.cs) base class and override in [/ChoiceInput.cs](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ChoiceInput.cs) to return the missing data.

## Testing

Before

After (choices are logged)

</blockquote>

Please review and, if necessary, port the changes.

### PR DESCRIPTION
**Pull request by ramfattah**
Fixes #4163 

## Description

This pull request fixes a bug for the multi-choice options not getting logged in application insights.

### CODE REVIEW

File: libraries/botbuilder-dialogs-adaptive/src/input/inputDialog.ts [hunk 1]
```diff
@@ -382,16 +382,31 @@ export abstract class InputDialog extends Dialog implements InputDialogConfigura
...
+
+        return msg;
+    }
+
+    /**
+     * @protected
+     * Track GeneratorResultEvent telemetry event with InputDialogResultEvent context.
+     * @param dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
+     * @param activityTemplate used to create the Activity.
+     * @param msg The [IMessageActivity](xref:botframework-schema.IMessageActivity) which will be sent.
```
Comment (EricDahlvang) [inline libraries/botbuilder-dialogs-adaptive/src/input/inputDialog.ts:395]: Should this comment be something regarding Partial [Activity](xref:botframework-schema.Activity) type, not IMessageActivity?

File: libraries/botbuilder-dialogs-adaptive/src/input/choiceInput.ts [hunk 2]
```diff
@@ -129,6 +132,28 @@ export class ChoiceInput extends InputDialog implements ChoiceInputConfiguration
...
+     * {@inheritDoc InputDialog.trackGeneratorResultEvent}
+     */
+    protected trackGeneratorResultEvent(
+        dc: DialogContext,
+        activityTemplate: TemplateInterface<Partial<Activity>, DialogStateManager>,
+        msg: Partial<Activity>
+    ): void {
+        const options = dc.state.getValue(ChoiceInput.OPTIONS_PROPERTY);
+        const properties: { [key: string]: string } = {
+            template: JSON.stringify(activityTemplate),
```
Comment (EricDahlvang) [inline libraries/botbuilder-dialogs-adaptive/src/input/choiceInput.ts:145]: Is JSON.stringify needed for these?

File: libraries/botbuilder-dialogs-adaptive/tests/choiceInputDialog.test.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
+const { ok, strictEqual } = require('assert');
+const { createTelemetryClientAndStub } = require('./telemetryUtils');
+const {
+    ConversationState,
+    MemoryStorage,
+    UserState,
+    AutoSaveStateMiddleware,
```
Comment (EricDahlvang) [inline libraries/botbuilder-dialogs-adaptive/tests/choiceInputDialog.test.js:7]: please remove unused requires

File: libraries/botbuilder-dialogs-adaptive/src/input/choiceInput.ts (hunk 2 above)
Comment (ramfattah) [reply libraries/botbuilder-dialogs-adaptive/src/input/choiceInput.ts:145]: I don't think it is needed, the properties in [trackGeneratorResultEvent ](https://github.com/microsoft/botbuilder-js/blob/main/libraries/botbuilder-dialogs-adaptive/src/input/inputDialog.ts#:~:text=%7D-,this.telemetryClient.trackEvent(%7B,%7D%2C,-%7D)%3B)function in base class are not using JSON.stringify. 

I tested without stringifying for these properties and updated the unit test and it passed.

thanks, good catch! making these changes.

**Review by EricDahlvang — APPROVED** [review]

**Review by johnataylor — APPROVED** [review]

**Pull request merged**